release/models/system/openconfig-system.yang

```diff
@@ -748,6 +754,14 @@ module openconfig-system {
       description
         "Source address to use on outgoing NTP packets";
     }
+
+    leaf key-id {
+      type leafref {
```

Comment:
I suggest we use the keychain model instead?
For example
```
    leaf keychain {
      type oc-keychain:keychain-ref;
      description
        "Reference to a keychain that should be used for NTP authentication.";
    }
```

Reply:
@Verma-Anukul thanks for the updates.  Can you update this to use the keychain model as well?

Reply:
@dplore  the suggestion here is to use keychain from openconfig-keychain.yang ? If so, ntp-keys defined in openconfig-system.yang won't be required any more right. 
And does any vendor reference for this, as I checked most of them uses ntp key configured specific for ntp model

Reply:
openconfig-keychain allows the use of many encryption algorithms whereas current NTP Authentication only allows MD5. As far as I can see the NTPv4 RFC only specifies MD5 as supported?

Reply:
@dplore a few concerns about using keychain

1. Not every customer will want to use keychain. We should allow them to use the keychain or the key-list under NTP.
2. The leaf we need is a reference to a key and thus cannot be a reference to a keychain. If the keychain is to be supported as an option, it needs to reference a specific key under a keychain.

Reply:
@dplore 
Please share your view

Reply:
Ok, we can treat this PR like a bug-fix, since this addition is needed to make the existing model paths for NTP usable.  We can address adding keychain support in a future PR.